### PR TITLE
Don't generate bounding box touching tiles in GeoTileGridAggregatorTests

### LINKED ISSUE #101077: [CI] GeoTileGridAggregatorTests testBoundedMultiValuedDocs failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/p4odiky5ufwii/tests/:server:test/org.elasticsearch.search.aggregations.bucket.geogrid.GeoTileGridAggregatorTests/testBoundedMultiValuedDocs
**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.geogrid.GeoTileGridAggregatorTests.testBoundedMultiValuedDocs" -Dtests.seed=5B6F7B1054EE4731 -Dtests.locale=no -Dtests.timezone=Africa/Maseru -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.aggregations.bucket.geogrid.GeoTileGridAggregatorTests&tests.test=testBoundedMultiValuedDocs
**Failure excerpt:**
```
java.lang.AssertionError: expected:<33> but was:<31>

  at __randomizedtesting.SeedInfo.seed([5B6F7B1054EE4731:F25BEDDD4E0FA215]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.failNotEquals(Assert.java:834)
  at org.junit.Assert.assertEquals(Assert.java:645)
  at org.junit.Assert.assertEquals(Assert.java:631)
  at org.elasticsearch.search.aggregations.bucket.geogrid.GeoGridAggregatorTestCase.lambda$testWithSeveralDocs$8(GeoGridAggregatorTestCase.java:154)
  at org.elasticsearch.search.aggregations.AggregatorTestCase.testCase(AggregatorTestCase.java:720)
  at org.elasticsearch.search.aggregations.bucket.geogrid.GeoGridAggregatorTestCase.testCase(GeoGridAggregatorTestCase.java:327)
  at org.elasticsearch.search.aggregations.bucket.geogrid.GeoGridAggregatorTestCase.testCase(GeoGridAggregatorTestCase.java:310)
  at org.elasticsearch.search.aggregations.bucket.geogrid.GeoGridAggregatorTestCase.testWithSeveralDocs(GeoGridAggregatorTestCase.java:153)
  at org.elasticsearch.search.aggregations.bucket.geogrid.GeoGridAggregatorTestCase.testBoundedMultiValuedDocs(GeoGridAggregatorTestCase.java:146)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

### PR DESCRIPTION
**Pull request by iverase**
in https://github.com/elastic/elasticsearch/pull/100826 we added a more aggresive way to handle geotile boundaries which has made this test fail whenever we gerate a bounding box that contains a share boundary with a tile. Therefore this PR make sure we generate bounding boxes that do not share boundaries with geotiles.

fixes https://github.com/elastic/elasticsearch/issues/101077

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/part-3

**Comment by iverase** [comment]
@elasticmachine update branch

File: test/framework/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregatorTestCase.java [hunk 1]
```diff
@@ -185,23 +186,24 @@ private void testWithSeveralDocs(BooleanSupplier supplier, GeoBoundingBox bbox)
...
 
     public void testMultiValuedDocsAsSubAgg() throws IOException {
-        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, null);
+        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, null, randomPrecision());
     }
 
     public void testBoundedMultiValuedDocsAsSubAgg() throws IOException {
-        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, randomBBox());
+        int precision = randomPrecision();
+        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, randomBBox(precision), randomPrecision());
```
Comment (craigtaverner) [inline test/framework/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregatorTestCase.java:203]: Why does this test generate two different random precisions, while `testNoundedSingletonDocsAsSubAgg` reuse the same precision twice?

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoTileGridAggregatorTests.java [hunk 2]
```diff
@@ -50,26 +52,49 @@ protected GeoBoundingBox randomBBox() {
...
             }
         );
-        // Avoid numerical errors for sub-atomic values
-        double left = GeoEncodingUtils.decodeLongitude(GeoEncodingUtils.encodeLongitude(bbox.left()));
-        double right = GeoEncodingUtils.decodeLongitude(GeoEncodingUtils.encodeLongitude(bbox.right()));
-        double top = GeoEncodingUtils.decodeLatitude(GeoEncodingUtils.encodeLatitude(bbox.top()));
-        double bottom = GeoEncodingUtils.decodeLatitude(GeoEncodingUtils.encodeLatitude(bbox.bottom()));
+        final int tiles = 1 << precision;
+
+        // Due to the way GeoTileBoundedPredicate works that adjust the given bounding box when it is touching the tiles, we need to
```
Comment (craigtaverner) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoTileGridAggregatorTests.java:57]: I'm slightly concerned that somehow we might be hiding issues, but cannot really argue that case yet.

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoTileGridAggregatorTests.java [hunk 3]
```diff
@@ -50,26 +52,49 @@ protected GeoBoundingBox randomBBox() {
...
+        // adjust here int order not to generate bounding boxes touching tiles or the test will fail
```
Comment (craigtaverner) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoTileGridAggregatorTests.java:58]: Spelling mistake

**Review by craigtaverner — APPROVED** [review]
Approved, although I have a nagging feeling we might be hiding potential isssues with this.

**Comment by craigtaverner** [comment]
This PR should also unmute the tests mutred in https://github.com/elastic/elasticsearch/pull/101109#issuecomment-1770369207

**Pull request merged**